presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -589,6 +589,23 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
         return hiveStatisticsProvider.getTableStatistics(session, ((HiveTableHandle) tableHandle).getSchemaTableName(), columns, columnTypes, partitions);
     }
 
+    @Override
+    public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle, Constraint<ColumnHandle> constraint, Set<ColumnHandle> desiredColumns)
```

Comment:
Let's don't change the SPI. You can simply request the stats using the old SPI, but before storing them in the `Input`, filter out the stats that are not referenced in the `TableScanNode` recreating the `TableStatistics` object.

Reply:
Requesting stats for only the columns needed requires more reworking in hive connector. By asking of filtering the stats for not used columns i just wanted to optimize the storage for the Input objects.